stories/Components/UIcomponents/Buttons/Switch/Switch.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import './switch.scss';
+import '../../../../Atom/Icons/icons.scss';
+import check from '../../../../assets/icons/checkmark-regular.svg';
+import cross from '../../../../assets/icons/disabled-x-regular.svg';
+
+export function Switch({
+    size = 'default',
+    state = 'default',
+    isToggled = true,
+    showIcon = false,
+    showLabel = false,
+    showValue = false,
+    onToggle,
+    label = 'Label:',
+    valueOn = 'On',
+    valueOff = 'Off',
+    ...props
+  }) {
+    const [toggled, setToggled] = useState(isToggled);
+  
+    useEffect(() => {
+        setToggled(isToggled);
+      }, [isToggled]);
+
+    const handleToggle = () => {
+      if (state !== 'disabled') {
+        setToggled(!toggled);
+        if (onToggle) onToggle(!toggled);
+      }
+    };
+  
+    const switchClass = `
+      switch
+      switch--${size}
+      switch--${state}
+      ${toggled ? 'switch--toggled' : 'switch--untoggled'}
+    `;
+
+    return (
+      <div className={switchClass} onClick={handleToggle} {...props}>
+        {showLabel && <div className="switch__label">{label}</div>}
+        <div className='switch__wrapper'>
+        <div className="switch__track" tabIndex={0}>
+            {showIcon && (
```

Comment:
To avoid using css filters (and hardcode color value) you might want to replace <img> element with <div>, which has respective .svg applied as a mask. 
Check Language switcher component's icons for the example of implementation